atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1817,11 +1840,15 @@ private void ensureUncommitted() {
     }
 
     private void ensureStillRunning() {
-        if (!(state.get() == State.UNCOMMITTED || state.get() == State.COMMITTING)) {
+        if (!isStillRunning()) {
```

Comment:
No test for this refactor aside from eye ball test

I don't want to make this VisibleForTesting as that would be potentially dangerous

If it's not okay to have this here due to safety, I can revert & not use `isStillRunning`; just duplicate the code

Reply:
would have been easier if we had `getState` as public or visible for testing